src/ejabberd_c2s.erl

```diff
@@ -2041,6 +2009,19 @@ presence_track(Acc, StateData) ->
                                    pres_a = A}}
     end.
 
+-spec process_presence_subscription_and_route(Acc :: mongoose_acc:t(),
+                                              Type :: subscribe | subscribed | unsubscribe | unsubscribed,
+                                              StateData :: state()) -> mongoose_acc:t().
+process_presence_subscription_and_route(Acc, Type, StateData) ->
+    From = mongoose_acc:from_jid(Acc),
+    User = StateData#state.user,
+    Server = StateData#state.server,
+    To = mongoose_acc:to_jid(Acc),
+    Acc1 = mongoose_hooks:roster_out_subscription(Server,
+                                                  Acc,
+                                                  User, To, Type),
+    check_privacy_and_route(Acc1, jid:to_bare(From), StateData).
```

Comment:
Is more readable and grepable:
```erlang
    Acc1 = mongoose_hooks:roster_out_subscription(Server, Acc, User, To, Type),
```

And generally, function calls on one line should be preferred. One way to reason to use several lines, if args are not fitting on one line. But even in this case we can do something like:

```erlang
    Acc1 = mongoose_hooks:resend_offline_messages_hook(
                                   StateData#state.server,
                                   Acc,
                                   StateData#state.user)
```
vs

```erlang
    #state{server = Server, user = User} = StateData,
    Acc1 = mongoose_hooks:resend_offline_messages_hook(Server, Acc, User)
```

Reply:
As for the first comment, I guess I was going for the consistency with how many hooks were written when they were using `ejabberd_hooks:run_fold` and wanted to leave a graphical indication of what are the arguments for the hook handler (but it is still not always obvious as many hooks are run with `Server` as one of the arguments). It also made it easier to manage long lines.
As for the second example, the formatting is weird in general there, I will fix this.